gtk4-macros/src/lib.rs

```diff
@@ -90,3 +94,90 @@ pub fn composite_template_derive(input: TokenStream) -> TokenStream {
     let gen = composite_template_derive::impl_composite_template(&input);
     gen.into()
 }
+
+/// Attribute macro for creating template callbacks from Rust methods.
+///
+/// The `template_callback` attribute is used to mark methods that will be
+/// exposed to the template scope. It can take the following options:
+///     - `name` which defaults to the function name if not defined
+///
+/// These methods can then be bound using the `<signal>` or `<closure>` tags
+/// in the template file. Note that the arguments and return type will only be
+/// checked at run time when the method is invoked.
+///
+/// Methods can optionally take `&self` as a first parameter. In this case, the
+/// attribute `swapped="true"` will usually have to be set on the `<signal>` or
+/// `<closure>` tag in order to invoke the function correctly.
+///
+/// # Example
+///
+///
+/// ```no_run
+/// # fn main() {}
+/// use gtk::prelude::*;
+/// use gtk::glib;
+/// use gtk::CompositeTemplate;
+/// use gtk::subclass::prelude::*;
+///
+/// mod imp {
+///     use super::*;
+///
+///     #[derive(Debug, Default, CompositeTemplate)]
+///     #[template(file = "test/template_callbacks.ui", callbacks)]
+///     pub struct MyWidget {
+///         #[template_child]
+///         pub label: TemplateChild<gtk::Label>,
+///         #[template_child(id = "my_button_id")]
+///         pub button: TemplateChild<gtk::Button>,
+///     }
+///
+///     #[glib::object_subclass]
+///     impl ObjectSubclass for MyWidget {
+///         const NAME: &'static str = "MyWidget";
+///         type Type = super::MyWidget;
+///         type ParentType = gtk::Box;
+///
+///         fn class_init(klass: &mut Self::Class) {
+///             Self::bind_template(klass);
+///         }
+///
+///         fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+///             obj.init_template();
+///         }
+///     }
+///
+///     #[gtk::composite_template_callbacks]
+///     impl MyWidget {
+///         #[template_callback]
```

Comment:
It would probably be useful to also allow something like

```rust
#[template_callback(name = "name of the callback in the .builder file")]
fn some_other_name() {}
```

Reply:
This is supported, I added a use of it to the composite_template example.